test/simple/test-child-process-stdout-flush-exit.js

```diff
@@ -52,14 +52,17 @@ if (process.argv[2] === 'child') {
 
   // check if we receive both 'hello' at start and 'goodbye' at end
   child.stdout.setEncoding('utf8');
+  var messageSequence = 0;
   child.stdout.on('data', function (data) {
-    if (data.slice(0, 6) == 'hello\n') {
+    if ((0 == messageSequence ) && (data.slice(0, 6) == 'hello\n')) {
```

Comment:
Styling comments:
* No space between values/variables and opening/closing parenthesis, so:
`if ((0 == messageSequence )` becomes `if ((0 == messageSequence)`.
* Having rvalues on the left of equality operators is not common practice in the code base, as far as I know, so `if ((0 == messageSequence )` would become `if ((messageSequence === 0)`.

Also, always use strict equality `===` and not loose equality `==` unless it's not possible.